conans/client/installer.py

```diff
@@ -289,6 +289,7 @@ def _build_package(self, export_folder, src_folder, build_folder, conan_file, ou
             output.info('Copying sources to build folder')
             shutil.copytree(src_folder, build_folder, symlinks=True)
         os.chdir(build_folder)
+        conan_file._conanfile_directory = build_folder
```

Comment:
I'd say this has been already done in another PR, maybe is the conflicting issue. Also, even if just two lines, they are a bit unrelated to this PR, so maybe deserve their own PR.